src/lib/summaryFilter.ts

```diff
@@ -0,0 +1,333 @@
+import { Matches } from "./comparison";
+import { Range } from "./range";
+import { RangesTuple } from "./summary";
+
+export class SummaryFilter {
```

Comment:
We should discuss the use of this filter object in person. The current implementation is not user-friendly because the "user" has to create this object as well as a summary object with the options split between them.